phases/19-capstone-projects/22-jsonrpc-stdio-transport/code/tests/test_transport.py

```
@@ -0,0 +1,159 @@
+
+class TestStreamDoesNotPoison(unittest.TestCase):
+    def test_parse_error_then_continue(self) -> None:
+        stdin = io.BytesIO()
+        stdin.write(b"{ broken\n")
+        stdin.write((json.dumps({"jsonrpc": "2.0", "id": 1, "method": "echo", "params": {"v": "ok"}}) + "\n").encode("utf-8"))
+        stdin.seek(0)
+        stdout = io.BytesIO()
+        transport = StdioTransport(stdin, stdout)
+        serve(echo_handler, transport)
+        stdout.seek(0)
+        lines = [json.loads(l) for l in stdout.read().decode("utf-8").splitlines() if l.strip()]
```

Ruff (0.15.14)

[error] Ambiguous variable name: `l`

(E741)